### ISSUE TITLE
Not able to compile the code on xcode 9.3

### ISSUE THREAD

**Issue by girijeshkumar2007** [Placenote/PlacenoteSDK-iOS#32]
Ld /Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Products/Debug-iphoneos/PlacenoteSDK/PlacenoteSDK.framework/PlacenoteSDK normal arm64
    cd /Users/girijeshkumar/Downloads/PlacenoteSDK-iOS-1.4.3/Pods
    export IPHONEOS_DEPLOYMENT_TARGET=11.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch arm64 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.3.sdk -L/Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Products/Debug-iphoneos/PlacenoteSDK -F/Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Products/Debug-iphoneos/PlacenoteSDK -F/Users/girijeshkumar/Downloads/PlacenoteSDK-iOS-1.4.3/Pods/PlacenoteSDK -filelist /Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/PlacenoteSDK.build/Objects-normal/arm64/PlacenoteSDK.LinkFileList -install_name @rpath/PlacenoteSDK.framework/PlacenoteSDK -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -miphoneos-version-min=11.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/PlacenoteSDK.build/Objects-normal/arm64/PlacenoteSDK_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos -Xlinker -add_ast_path -Xlinker /Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/PlacenoteSDK.build/Objects-normal/arm64/PlacenoteSDK.swiftmodule -framework Placenote -framework Foundation -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/PlacenoteSDK.build/Objects-normal/arm64/PlacenoteSDK_dependency_info.dat -o /Users/girijeshkumar/Library/Developer/Xcode/DerivedData/PlacenoteSDKExample-gzttriufetdqqnclcqfucdvnsiwx/Build/Products/Debug-iphoneos/PlacenoteSDK/PlacenoteSDK.framework/PlacenoteSDK

ld: warning: ignoring file /Users/girijeshkumar/Downloads/PlacenoteSDK-iOS-1.4.3/Pods/PlacenoteSDK/Placenote.framework/Placenote, file was built for unsupported file format ( 0x76 0x65 0x72 0x73 0x69 0x6F 0x6E 0x20 0x68 0x74 0x74 0x70 0x73 0x3A 0x2F 0x2F ) which is not the architecture being linked (arm64): /Users/girijeshkumar/Downloads/PlacenoteSDK-iOS-1.4.3/Pods/PlacenoteSDK/Placenote.framework/Placenote
Undefined symbols for architecture arm64:
  "_PNSetFrame", referenced from:
      _setFrameNative in LibPlacenote.o
  "_PNGetPose", referenced from:
      _getPoseNative in LibPlacenote.o
  "_PNSetIntrinsics", referenced from:
      _setIntrinsicsNative in LibPlacenote.o
  "_PNInitialize", referenced from:
      _initializeSDK in LibPlacenote.o
  "_PNStartRecordDataset", referenced from:
      PlacenoteSDK.LibPlacenote.startReportRecord(uploadProgressCb: (Swift.Bool, Swift.Bool, Swift.Float) -> ()) -> () in LibPlacenote.o
  "_PNSaveMap", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.PNCallbackResult_t>?, Swift.UnsafeMutableRawPointer?) -> () in PlacenoteSDK.LibPlacenote.saveMap(savedCb: (Swift.String?) -> (), uploadProgressCb: (Swift.Bool, Swift.Bool, Swift.Float) -> ()) -> () in LibPlacenote.o
  "_PNAddMap", referenced from:
      PlacenoteSDK.LibPlacenote.saveMap(savedCb: (Swift.String?) -> (), uploadProgressCb: (Swift.Bool, Swift.Bool, Swift.Float) -> ()) -> () in LibPlacenote.o
  "_PNListMaps", referenced from:
      PlacenoteSDK.LibPlacenote.fetchMapList(listCb: (Swift.Bool, [Swift.String : Any]) -> ()) -> () in LibPlacenote.o
  "_PNStopSession", referenced from:
      PlacenoteSDK.LibPlacenote.stopSession() -> () in LibPlacenote.o
  "_PNDeleteMap", referenced from:
      PlacenoteSDK.LibPlacenote.deleteMap(mapId: Swift.String, deletedCb: (Swift.Bool) -> ()) -> () in LibPlacenote.o
  "_PNSetMetadata", referenced from:
      PlacenoteSDK.LibPlacenote.setMapMetadata(mapId: Swift.String, metadataJson: Swift.String) -> Swift.Bool in LibPlacenote.o
  "_PNGetStatus", referenced from:
      PlacenoteSDK.LibPlacenote.getMappingStatus() -> PlacenoteSDK.LibPlacenote.MappingStatus in LibPlacenote.o
  "_PNGetAllLandmarks", referenced from:
      PlacenoteSDK.LibPlacenote.getAllLandmarks() -> [__C.PNFeaturePoint_t] in LibPlacenote.o
  "_PNGetTrackedLandmarks", referenced from:
      PlacenoteSDK.LibPlacenote.getTrackedLandmarks() -> [__C.PNFeaturePoint_t] in LibPlacenote.o
  "_PNLoadMap", referenced from:
      PlacenoteSDK.LibPlacenote.loadMap(mapId: Swift.String, downloadProgressCb: (Swift.Bool, Swift.Bool, Swift.Float) -> ()) -> () in LibPlacenote.o
  "_PNStartSession", referenced from:
      PlacenoteSDK.LibPlacenote.startSession() -> () in LibPlacenote.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by pmukherj** [comment]
Hey @girijeshkumar2007. 

Sorry you are having this trouble. This is most likely because you are missing some of the library files. These are large files and therefore stored in git lfs. You have two options here: 
a) We have some instructions to install git lfs in the github readme:
To Install these files, install lfs either using HomeBrew:
brew install git-lfs
or MacPorts:
port install git-lfs
And then, to get the library files, run:
git lfs install
git lfs pull

or 
b)You can get a complete tar ball with everything included right here: https://github.com/Placenote/PlacenoteSDK-iOS/releases/download/v1.4.3/PlacenoteSDKExample.tar.gz

**Comment by y22ma** [comment]
@girijeshkumar2007 just checking if this resolved your issue?

**Comment by Jordan-Campbell** [comment]
I'm having this same error when using cocoapods. The files all appear to be retrieved correctly, however it can't seem to link against them. I'm building for a valid device etc. Everything works fine if I add Placenote manually.

**Comment by pmukherj** [comment]
hey @Jordan-Campbell,

Just checking that by "files all appear to be retrieve correctly", is Placenote.framework file is 7-9MBish?

**Comment by Jordan-Campbell** [comment]
Ok no, it's not! It's only 25KB.

**Comment by pmukherj** [comment]
Right, so that's why its not able to link it. The main binary is the Placenote.framework/Placenote and it should be atleast 7.4MB. 

We've seen this problem before and its due to some configuration issue within CocoaPods on *some* setups. Please copy-paste for now, we are working on a fix!

**Comment by pmukherj** [comment]
Thanks! This is helpful.
